packages/wallets/src/features/cashier/flows/WalletTransfer/WalletTransfer.tsx

```diff
@@ -1,11 +1,27 @@
 import React, { useCallback, useEffect } from 'react';
-import { useTransferBetweenAccounts } from '@deriv/api-v2';
+import { useActiveWalletAccount, useAuthorize, useTransferBetweenAccounts } from '@deriv/api-v2';
 import { Loader } from '../../../../components';
 import { TransferModule } from '../../modules';
 import { TransferNotAvailable } from '../../screens/TransferNotAvailable';
 
 const WalletTransfer = () => {
     const { data, isLoading: isTransferAccountsLoading, mutate } = useTransferBetweenAccounts();
+    const { switchAccount } = useAuthorize();
+    const { data: activeWallet } = useActiveWalletAccount();
+
+    useEffect(() => {
+        const queryParams = new URLSearchParams(location.search);
+        const loginidQueryParam = queryParams.get('loginid');
+
+        // if loginid query param doesn't match active wallet's loginid on mount, initiate account switching
+        if (loginidQueryParam && loginidQueryParam !== activeWallet?.loginid) {
```

Comment:
Given the "?" in `activeWallet?.loginid`, would the switching get initiated every time `activeWallet?.loginid` is `undefined`? Wouldn't this in turn create an infinite loop or smth? I suggest checking whether `activeWallet?.loginid` is not `undefined` as a part of the condition for account switching

Reply:
Moved logic to WalletsAuthProvider instead